mmf/modules/encoders.py

```diff
@@ -22,6 +24,12 @@
 from transformers.modeling_auto import AutoModel
 
 
+try:
+    from detectron2.modeling import ShapeSpec, build_resnet_backbone
+except ImportError:
+    pass
```

Comment:
It would make more sense to import it below where it is actually required. In case it is missing then, it would be to tell the user that detectron2 needs to be installed before using that particular resnet encoder.

Reply:
I had tried that earlier. Moving it inside the encoder class causes an error in detectron2's call to fvcore, where it tries to jit compile an unrelated `sigmoid_focal_loss` function and throws error. I will need more time to investigate why that is occuring.

```
torch.jit.frontend.NotSupportedError: Compiled functions can't take variable number of arguments or use keyword-only arguments with defaults
```